lib/literature/uploaders.ex

```diff
@@ -40,7 +40,21 @@ defmodule Literature.Uploaders do
     {:convert, "-format avif", :avif}
   end
 
-  def storage_dir(_, _), do: "literature"
+  def storage_dir(:original, {file, _}) do
+    file_name = Path.basename(file.file_name, Path.extname(file.file_name))
+
+    "literature/#{file_name}"
+  end
+
+  def storage_dir(version, {file, _}) do
+    file_name =
+      file.file_name
+      |> Path.basename(Path.extname(file.file_name))
+      |> String.split("-")
+      |> List.delete_at(-1)
+
+    "literature/#{file_name}/#{version}"
```

Comment:
can we go with a uuid or something more unique to avoid possible duplicates over time

Reply:
I update the path with literature/{{publication_id}}/{{post_slug}}/{{version}}